### PR TITLE
Updated APIs to match other language clients and opensearch openapi spec

### LINKED ISSUE #499: APIs with Naming differences (Generated vs Existing Python Client)
Module | existing client has | generated client has
-- | -- | --
Async/__init__ | list_all_point_in_time | get_all_pits
Async/__init__ | create_point_in_time | create_pit
Async/security | health_check | health
Async/security | update_audit_config | update_audit_configuration

### PR DESCRIPTION
**Pull request by saimedhi**
### Description
Updated APIs to match other language clients and opensearch openapi spec. APIs added to "_patch" are manually written. Other APIs added are Generated using openapispec.

Deprecated APIs: They are moved into client/_patch.py
list_all_point_in_time, create_point_in_time, health_check, update_audit_config, delete_point_in_time

Newly Added APIs: They are moved into corresponding namespaces
get_all_pits, create_pit, health, update_audit_configuration, delete_all_pits, delete_pit

### Issues Resolved
Closes #499, #500, #501

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: opensearchpy/_async/client/__init__.py [hunk 1]
```diff
@@ -1955,64 +1961,66 @@ async def get_script_languages(self, params=None, headers=None):
             "GET", "/_script_language", params=params, headers=headers
         )
 
-    @query_params()
-    async def list_all_point_in_time(self, params=None, headers=None):
+    async def create_pit(self, index, params=None, headers=None):
```
Comment (VachaShah) [inline opensearchpy/_async/client/__init__.py:1964]: Looks like this is missing query_params?

File: opensearchpy/_async/client/_patch.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+import warnings
+
+from .utils import SKIP_IN_PATH, query_params
+
+
+@query_params()
+async def list_all_point_in_time(self, params=None, headers=None):
```
[the remark this answers is not in the document]
Comment (saimedhi) [reply opensearchpy/_async/client/_patch.py:16]: yes

File: opensearchpy/_async/client/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (saimedhi) [reply opensearchpy/_async/client/__init__.py:1964]: generated

**Comment by saimedhi** [comment]
Hi @florianvazelle, I would greatly appreciate it if you could kindly review this PR and share your feedback whenever it's convenient for you. Thank you!

**Review by VachaShah — APPROVED** [review]
This looks great! Can we also start adding the generation details in a markdown file? For example, what the patch file is supposed to be? How the generator works etc? I see there is already https://github.com/opensearch-project/opensearch-py/blob/main/DEVELOPER_GUIDE.md#running-python-client-generator

**Comment by saimedhi** [comment]
> This looks great! Can we also start adding the generation details in a markdown file? For example, what the patch file is supposed to be? How the generator works etc? I see there is already https://github.com/opensearch-project/opensearch-py/blob/main/DEVELOPER_GUIDE.md#running-python-client-generator

@VachaShah, I'll include these details in my upcoming PR. I'll create a markdown file explaining the generator's functionality and the client structure, including what's placed in patch files. You can find most of the information [here](https://github.com/opensearch-project/opensearch-py/issues/477#issue-1853894225), and I'll ensure it's added to the repository soon.

**Comment by saimedhi** [comment]
@VachaShah, Shall we merge this

**Review by dblock — APPROVED** [review]
Looks good.

For generated files we need some kind of "this is generated" header.

**Pull request merged**